data/transactions/logic/evalStateful_test.go

```diff
@@ -1416,6 +1413,20 @@ intc_1
 	testApp(t, source, now, "cannot compare ([]byte to uint64)")
 }
 
+func TestAppParams(t *testing.T) {
+	t.Parallel()
+	ep, ledger := makeSampleEnv()
+	ledger.newAccount(ep.Txn.Txn.Sender, 1)
+	ledger.newApp(ep.Txn.Txn.Sender, 100, basics.AppParams{})
+
+	/* app id is in ForeignApps, but does not exist */
+	source := "int 56; app_params_get AppExtraProgramPages; int 0; ==; assert; int 0; =="
+	testApp(t, source, ep)
+	/* app id is in ForeignApps, but has zero ExtraProgramPages */
+	source = "int 100; app_params_get AppExtraProgramPages; int 1; ==; assert; int 0; =="
```

Comment:
could you pleases add a test for `app_params_get` of different approval and clear state programs? Just because Jason spotted it and this would prevent further copy/paste errors on these fields in future

Reply:
Ok, all params are tested in e2e subs test now. (teal-app-params.sh)